### ISSUE TITLE
Swig tests fail

### ISSUE THREAD

**Issue by lundj** [numpy/numpy#15743]
The included swig tests in tools/swig/ fails

### Reproducing code example:

```bash
cd tools/swig/
make test
```

### Error message:

ERROR: testResizeBad1 (__main__.Array2TestCase)
Test Array2 resize method, negative nrows

Traceback (most recent call last):
  File "testArray.py", line 188, in testResizeBad1
    self.assertRaises(ValueError, self.array2.resize, -5, 5)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 473, in assertRaises
    callableObj(*args, **kwargs)
TypeError: resize() takes exactly 2 arguments (3 given)

ERROR: testResizeBad2 (__main__.Array2TestCase)
Test Array2 resize method, negative ncols
Traceback (most recent call last):
  File "testArray.py", line 192, in testResizeBad2
    self.assertRaises(ValueError, self.array2.resize, 5, -5)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 473, in assertRaises
    callableObj(*args, **kwargs)
TypeError: resize() takes exactly 2 arguments (3 given)

Ran 58 tests in 0.007s
FAILED (errors=3)
make[1]: *** [test] Error 1
make: *** [test] Error 2

### Numpy/Python version information:
Run on Ubuntu 19 with gcc 7.5, python 3.6.9. Tried for  numpy versions 1.14.6, 1.16.6, 1.17.0, and 1.18.1
Same result on mac 10.15.3 with clang 11.0, python 2.7.16 and all numpy versions.

**Comment by charris** [comment]
Reproduced on Fedora 31 with Python 3.7.

It is strange that the problem has been around for so long undetected, looks like a genuine error in `swig.i`. I suspect the functions are missing the bound array.

**Comment by charris** [comment]
@mlamarre Thoughts?

**Comment by charris** [comment]
Could very well be an error in the test files.

**Comment by charris** [comment]
I'm starting to think this may be a swig bug. The function is correctly defined in `Array2.h`, but the generated wrapper function looks like the `Array1` version. Note that `Array1.h` is included in `Array2.h` and both define `resize` methods. I'm thinking that maybe swig is loosing track of which goes where. A swig bug might also explain why the error has gone unnoticed for so many numpy releases when very little has changed in the files.

I have SWIG Version 4.0.1, what version are you using?

**Comment by charris** [comment]
Also fails with NumPy 1.8.0 and SWIG Version 3.0.12.

**Comment by charris** [comment]
Looking like a bug in the tests.

**Comment by mlamarre** [comment]
The following typemap interferes:

https://github.com/numpy/numpy/blob/1cad7f83b1fd0c1f7afab86b0afdf4536b6f3e75/tools/swig/test/Array.i#L51

with the following signature:

https://github.com/numpy/numpy/blob/a38888c18cd2a20de0eb0578b3fa8660cda79582/tools/swig/test/Array2.h#L35

**Comment by mlamarre** [comment]
I have it fix will submit PR in a few minutes

**Comment by charris** [comment]
@mlamarre How? I notice that the python interface is incorrect, or at least takes only two arguments.
```
class Array2(object):
...
    def resize(self, nrows):
        return _Array.Array2_resize(self, nrows)
...
```
but I don't know why :) If I add arguments the failing tests pass, but new ones fail. I can't see why the tests ever passed.

**Comment by mlamarre** [comment]
I think the tests didn't pass in a long while. 
https://github.com/numpy/numpy/pull/15752

**Comment by mlamarre** [comment]
Two hypothesis: 91f083a694e9c9185b6a2e10275a0616d18064a6 fixed testResize1 but broke the other tests or maybe it's a regression in Swig related to default values mixed with typemaps. I don't know.

**Comment by charris** [comment]
Fixed by #15752.
